bifrost/app/price-calc/ModelPriceCalculator.tsx

```diff
@@ -0,0 +1,353 @@
+"use client";
+
+import { useState, useEffect } from "react";
+import { Table, Zap } from "lucide-react";
+import { costOf, costOfPrompt } from "../../packages/cost"; // Ensure the path is correct
+import { providers } from "../../packages/cost/providers/mappings"; // Ensure the path is correct
+import { useRouter } from "next/navigation";
+import Image from "next/image";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+
+// Function to format provider names
+function formatProviderName(provider: string): string {
+  const formattingMap: { [key: string]: string } = {
+    OPENAI: "OpenAI",
+    ANTHROPIC: "Anthropic",
+    AZURE: "Azure",
+    TOGETHER: "Together AI",
+    FIREWORKS: "Fireworks",
+    OPENROUTER: "OpenRouter",
+    GROQ: "Groq",
+    QSTASH: "Qstash",
+    MISTRAL: "Mistral",
+  };
+
+  return formattingMap[provider.toUpperCase()] || provider.toUpperCase();
+}
+
+type ModelPriceCalculatorProps = {
+  model: string;
+  provider: string;
+};
+
+type CostData = {
+  provider: string;
+  model: string;
+  inputCostPer1k: number;
+  outputCostPer1k: number;
+  inputCost: number;
+  outputCost: number;
+  totalCost: number;
+};
+
+// Function to sanitize model names for URL
+function sanitizeModelName(name: string): string {
+  return name.replace(/[\/\\]/g, "-");
+}
+
+// Update this function to include the provider in the URL
+function sanitizeForUrl(name: string): string {
+  return name.replace(/[\/\\]/g, "-").toLowerCase();
+}
+
+export default function ModelPriceCalculator({
```

Comment:
Update the model price calculator to look like: https://www.figma.com/design/JbVs4mCPz4f5KfMuh2XEy2/Helicone-Landing-Page?node-id=2380-1532&node-type=frame&m=dev